[libc] Output all headers with LIBC_CONF_OUTPUT_ALL_HEADERS (#144114)
llvm/llvm-project: libc/CMakeLists.txt

option(LLVM_LIBC_FULL_BUILD "Build and test LLVM libc as if it is the full libc" ${default_to_full_build})
option(LLVM_LIBC_IMPLEMENTATION_DEFINED_TEST_BEHAVIOR "Build LLVM libc tests assuming our implementation-defined behavior" ON)
option(LLVM_LIBC_ENABLE_LINTING "Enables linting of libc source files" OFF)

> option(LLVM_LIBC_ALL_HEADERS "Outputs all functions in header files, regardless of whether it has been implemented" OFF)

Reviewer: nit:
```suggestion
option(LLVM_LIBC_ALL_HEADERS "Outputs all functions in header files, regardless of whether they are enabled on this target" OFF)
```
